### PR TITLE
vendored externals: update archspec

### PR DESCRIPTION
**Pull request by alalazo**
Modifications:
- [x] Better support for a64fx
- [x] Better support for Apple M1

### CODE REVIEW

File: lib/spack/external/__init__.py [hunk 1]
```diff
@@ -18,7 +18,7 @@
 
 * Homepage: https://pypi.python.org/pypi/archspec
 * Usage: Labeling, comparison and detection of microarchitectures
-* Version: 0.1.2 (commit 85757b6666422fca86aa882a769bf78b0f992f54)
+* Version: 0.1.2 (commit 1d616ad3dcd27e8a90cd5f4c8914ee233a186b0c)
```
Comment (haampie) [inline lib/spack/external/__init__.py]: Why don't you make a new release?

Comment (alalazo) [reply lib/spack/external/__init__.py]: Trying to see if we can fit also some improvements to `graviton`. In any case I think we'll release in time for ISC.

**Comment by adamjstewart** [comment]
Tested this on macOS 12.3.1 and Apple M1 Pro. On both develop and this branch, I see multiple error messages and an incorrect architecture determination:
```console
$ spack arch
sysctl: unknown oid 'machdep.cpu.leaf7_features'
sysctl: unknown oid 'machdep.cpu.vendor'
sysctl: unknown oid 'machdep.cpu.model'
darwin-monterey-x86_64
```
EDIT: Opened an issue https://github.com/archspec/archspec/issues/83

**Comment by alalazo** [comment]
This now contains the fix to https://github.com/archspec/archspec/issues/83

**Review by adamjstewart — APPROVED** [review]
Works now, thanks @alalazo!

**Pull request merged**